soc/silabs/silabs_s2/xg22/Kconfig.soc

```diff
@@ -13,12 +13,52 @@ config SOC_SERIES_EFR32BG22
 	help
 	  Silicon Labs EFR32BG22 (Blue Gecko) Series MCU
 
+config SOC_SERIES_EFR32MG22
+	bool
+	select SOC_SILABS_XG22
+	help
+	  Silicon Labs EFR32MG22 (Mighty Gecko) Series MCU
+
 config SOC_PART_NUMBER_EFR32BG22C224F512IM40
 	bool
 	select SOC_SERIES_EFR32BG22
 
+config SOC_PART_NUMBER_EFR32MG22C224F512IM40
+	bool
+	select SOC_SERIES_EFR32MG22
+
+config SOC_PART_NUMBER_EFR32MG22C224F512IM32
+	bool
+	select SOC_SERIES_EFR32MG22
+
+config SOC_PART_NUMBER_EFR32MG22C224F512GN32
+	bool
+	select SOC_SERIES_EFR32MG22
+
+config SOC_PART_NUMBER_EFR32MG22E224F512IM40
+	bool
+	select SOC_SERIES_EFR32MG22
+
+config SOC_PART_NUMBER_EFR32MG22E224F512IM32
+	bool
+	select SOC_SERIES_EFR32MG22
+
 config SOC_SERIES
 	default "efr32bg22" if SOC_SERIES_EFR32BG22
+	default "efr32mg22" if SOC_SERIES_EFR32MG22
 
 config SOC
 	default "efr32bg22c224f512im40" if SOC_PART_NUMBER_EFR32BG22C224F512IM40
+	default "efr32mg22c224f512im40" if SOC_PART_NUMBER_EFR32MG22C224F512IM40
+	default "efr32mg22c224f512im32" if SOC_PART_NUMBER_EFR32MG22C224F512IM32
+	default "efr32mg22c224f512gn32" if SOC_PART_NUMBER_EFR32MG22C224F512GN32
+	default "efr32mg22e224f512im40" if SOC_PART_NUMBER_EFR32MG22E224F512IM40
+	default "efr32mg22e224f512im32" if SOC_PART_NUMBER_EFR32MG22E224F512IM32
+
+config SOC_PART_NUMBER
```

Comment:
This option was removed in https://github.com/zephyrproject-rtos/zephyr/pull/93207

Reply:
Removed